tests/test_main.py

```diff
@@ -471,3 +471,93 @@ def test_datarobot_transferable_call(monkeypatch):
             compression=False,
             field_size_limit=None,
         )
+
+
+def test_resume_key(monkeypatch):
```

Comment:
I think it might be a bit simpler and clear if you split this test into `test_resume_option` and `test_no_resume_option`.

It'd be also nice to add another test case that tests our *resume* behavior deep inside, by detecting which context exactly is created.